os/kernel/task_monitor/task_monitor.c

```diff
@@ -19,16 +19,63 @@
  * Included Files
  ****************************************************************************/
 #include <tinyara/config.h>
+
 #include <sched.h>
+#include <queue.h>
+#include <semaphore.h>
+
 #include <sys/boardctl.h>
 #include <tinyara/sched.h>
+
 #include "task_monitor_internal.h"
 
-static int monitored_tasks_list[CONFIG_MAX_TASKS];
+static taks_monitor_t g_monitored_tasks_list[CONFIG_MAX_TASKS];
+static taks_monitor_que_t g_que_list[CONFIG_TASK_MONITOR_MAX_INTERVAL / CONFIG_TASK_MONITOR_INTERVAL];
+static int g_monitor_cnt;
+static sem_t g_stop_sem;
 
-void task_monitor_update_list(int pid, int status)
+void task_monitor_register_list(int pid, int interval)
 {
-	monitored_tasks_list[PIDHASH(pid)] = status;
+	if (interval < CONFIG_TASK_MONITOR_INTERVAL && CONFIG_TASK_MONITOR_MAX_INTERVAL < interval) {
+		dbg("Must be greater than %d and less than or equal to %d.\n", CONFIG_TASK_MONITOR_INTERVAL, CONFIG_TASK_MONITOR_MAX_INTERVAL);
```

Comment:
For this, let's add return type for this function.

Reply:
done